parquet-column/src/main/java/org/apache/parquet/column/values/RequiresPreviousReader.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values;
+
+public interface RequiresPreviousReader {
```

Comment:
Not sure if this interface/generation is actually needed.
The actual problem is the DeltaByteArrayReader needs to reset the 'previous' member to fix the bug.
It needs the previous char array, the previous value, not the previous reader. This interface gives an impression the subclass could hold a reference of the previous reader.

How about:
remove this interface,
create a setPreviousValue method in DeltaByteArrayReader.
and call:
setPreviousValue(deltaByteArrayReader.previous)